src/main/java/org/asciidoc/intellij/actions/asciidoc/ImageService.java

```diff
@@ -0,0 +1,52 @@
+package org.asciidoc.intellij.actions.asciidoc;
+
+import com.intellij.notification.Notification;
+import com.intellij.notification.NotificationType;
+import com.intellij.notification.Notifications;
+import com.intellij.openapi.vfs.VirtualFile;
+import org.asciidoc.intellij.editor.AsciiDocPreviewEditor;
+
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+import java.io.IOException;
+import java.util.concurrent.CompletableFuture;
+
+public final class ImageService {
+
+  private static final int FALLBACK_IMAGE_WIDTH = 0;
```

Comment:
A fallback image width of 0 leads to an image that doesn't show in the document. This is a surprise for the user that I want to avoid.

Could the default image with be "empty", leading to an empty input field instead of 0 in the UI? Instead of an "int", maybe it could be an "Optional<Integer>" or an "Integer" that can be "null"?

Reply:
I like the idea of using an empty Optional here :+1: